api/src/main/java/com/okta/idx/sdk/api/wrapper/AuthenticationWrapper.java

```diff
@@ -164,72 +184,617 @@ public static AuthenticationResponse authenticate(IDXClient client, Authenticati
+            logger.error("Something went wrong! {}, {}", e, authenticationResponse.getErrors());
+        } catch (IllegalArgumentException e) {
+            logger.error("Exception occurred", e);
+            authenticationResponse.addError(e.getMessage());
+        }
+
+        authenticationResponse.setIdxClientContext(idxClientContext);
+        return authenticationResponse;
+    }
+
+    /**
+     * Helper to populate the UI options to be shown on Authenticator options page.
+     *
+     * @param client                the IDX Client
+     * @param idxClientContext      the IDC Client context
+     * @return the list of {@link AuthenticatorUIOption} options
+     */
+    public static List<AuthenticatorUIOption> populateAuthenticatorUIOptions(IDXClient client, IDXClientContext idxClientContext) {
+
+        List<AuthenticatorUIOption> authenticatorUIOptionList = new LinkedList<>();
+
+        try {
+            IDXResponse introspectResponse = client.introspect(idxClientContext);
+
+            RemediationOption[] remediationOptions = introspectResponse.remediation().remediationOptions();
+            printRemediationOptions(remediationOptions);
+
+            RemediationOption remediationOption = extractRemediationOption(remediationOptions, RemediationType.SELECT_AUTHENTICATOR_ENROLL);
+
+            Map<String, String> authenticatorOptions = remediationOption.getAuthenticatorOptions();
+
+            for (Map.Entry<String, String> entry : authenticatorOptions.entrySet()) {
+                authenticatorUIOptionList.add(new AuthenticatorUIOption(entry.getValue(), entry.getKey()));
+            }
+        } catch (Exception e) {
+            logger.error("Error occurred:", e);
+        }
+
+        return authenticatorUIOptionList;
+    }
+
+    /**
+     * Helper to check if we have landed terminal success/no more remediation steps to follow.
+     *
+     * @param client                the IDX Client
+     * @param idxClientContext      the IDC Client context
+     * @return true if login is successful and if there are no more remediation steps to follow; false otherwise.
+     */
+    public static boolean isTerminalSuccess(IDXClient client, IDXClientContext idxClientContext) {
+        try {
+            return client.introspect(idxClientContext).isLoginSuccessful();
+        } catch (ProcessingException e) {
+            logger.error("Error occurred:", e);
+            return false;
+        }
+    }
+
+    /**
+     * Helper to check if we have optional authenticators to skip in current remediation step.
+     *
+     * @param client                the IDX Client
+     * @param idxClientContext      the IDC Client context
+     * @return true if we have optional authenticators to skip; false otherwise.
+     */
+
+    public static boolean isSkipAuthenticatorPresent(IDXClient client, IDXClientContext idxClientContext) {
```

Comment:
Is this something that can be exposed on the response instead? We should get this back without having to call introspect.

Reply:
The common theme is to call `introspect` with the available `idxClientContext` instead of directly processing /relying the previous step's response. So, I'll stick to that theme here also. We can look at optimizing this by avoiding an extra API call here in the future (as this is a common pattern implemented across backend SDKs).